pybamm/settings.py

```diff
@@ -43,35 +43,49 @@ def simplify(self, value):
         self._simplify = value
 
     def set_smoothing_parameters(self, k):
-        "Helper function to set all smoothing parameters"
-        self.min_smoothing = k
-        self.max_smoothing = k
+        """Helper function to set all smoothing parameters"""
+        if k == "exact":
+            self.min_max_mode = "exact"
+        else:
+            self.min_max_smoothing = k
+            self.min_max_mode = "soft"
```

Comment:
Is the idea that we will deprecate this in the future, as this is redundant functionality at the moment? If so, can you update [this notebook](https://docs.pybamm.org/en/latest/source/examples/notebooks/solvers/speed-up-solver.html) to use the new syntax?

Reply:
Also, summoning @tinosulzer here to get his input on the API.

Reply:
I was not sure what the plan was for the soft and smooth min/max functions. It seems like the soft versions are tied to the abs and heaviside functions as well. I did not want to change too much in the scope of this ticket. I essentially implemented it as a hidden feature until I know more.

I think similar changes should be made for the other smoothed functions. Especially since the value of k is effectively assumed to match for all of them. However, the new smoothing function needs a k value 2 orders of magnitude higher than the old soft version. So if this becomes the preferred smoothing method then other parts of the API should be adjusted.

Reply:
I updated that notebook, but it appears to be failing even on develop. I will investigate that with fresh eyes this weekend.

Reply:
> I updated that notebook, but it appears to be failing even on develop. I will investigate that with fresh eyes this weekend.

I confirmed that all failing cells were intended to fail. Just panicked for a second when I thought I broke a notebook.